src/mesh/python/mesh_types.py

```diff
@@ -287,6 +287,319 @@ def __init__(self, bounds_per_dim, num_cells_per_dim):
                                nodes_per_side_zlow, nodes_per_side_zhig]
 
 
+# ------------------------------------------------------------------------------------------------ #
+# Voronoi 2D mesh type
+class vor_2d_mesh(base_mesh):
```

Comment:
Nice!  This looks pretty involved.

Reply:
A lot more work than I thought it was going to be...